### PR TITLE
Fix /index.json route to pass the format

### PR DESCRIPTION
**Pull request by coolo**
/index.json matches the route but doesn't know it has to reply with json.
So just use /index with given formats

### CODE REVIEW

File: lib/OpenQA/WebAPI.pm [hunk 1]
```diff
@@ -216,9 +216,9 @@ sub startup {
 
     # Favicon
     $r->get('/favicon.ico' => sub { my $c = shift; $c->render_static('favicon.ico') });
+    $r->get('/index' => [format => ['html', 'json']])->to('main#index');
```
Comment (Martchus) [inline lib/OpenQA/WebAPI.pm]: I think it is not necessary to specify the formats explicitly: `$r->get('/index')->to('main#index');`

Comment (coolo) [inline lib/OpenQA/WebAPI.pm]: but I want it not to reply to .txt :)

Comment (Martchus) [inline lib/OpenQA/WebAPI.pm]: Ok. With my version, if you enter `index.txt` you get back to `index`. With your version you get 404 in this case.

Comment (coolo) [inline lib/OpenQA/WebAPI.pm]: My initial version only supported /index.json - but I wanted to also support index.html to avoid confusion

**Comment by Martchus** [comment]
Can you add this test (and fill the TODO)?
```
diff --git a/t/ui/14-dashboard.t b/t/ui/14-dashboard.t
index a0ff692..4ccf4e4 100644
--- a/t/ui/14-dashboard.t
+++ b/t/ui/14-dashboard.t
@@ -88,5 +88,8 @@ subtest 'filter form' => sub {
     is($driver->get_current_url(), $baseurl . '?group=SLE+12+SP2&limit_builds=38&time_limit_days=142#', 'URL parameters for filter are correct');
 };
 
+$driver->get($baseurl . '/index.json');
+is($driver->get_page_source(), 'TODO', 'page rendered as JSON');
+
 t::ui::PhantomTest::kill_phantom();
 done_testing();
```

**Comment by coolo** [comment]
that was a trick question, right? :)

**Comment by Martchus** [comment]
These 2 lines test whether actually JSON is responded which fails on the current master but should work with the proposed fix applied.

**Comment by coolo** [comment]
test added

**Pull request merged**